graphql-dgs-spring-webmvc-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/webmvc/autoconfigure/DgsWebMvcConfigurationProperties.kt

```diff
@@ -19,31 +19,51 @@ package com.netflix.graphql.dgs.webmvc.autoconfigure
 import org.springframework.boot.context.properties.ConfigurationProperties
 import org.springframework.boot.context.properties.ConstructorBinding
 import org.springframework.boot.context.properties.bind.DefaultValue
+import org.springframework.validation.annotation.Validated
+import javax.validation.Constraint
+import javax.validation.Valid
+import javax.validation.constraints.Pattern
+import kotlin.reflect.KClass
 
 /**
  * Configuration properties for DGS web controllers.
  */
 @ConstructorBinding
 @ConfigurationProperties(prefix = "dgs.graphql")
+@Validated
 @Suppress("ConfigurationProperties")
 data class DgsWebMvcConfigurationProperties(
     /** Path to the GraphQL endpoint without trailing slash. */
-    @DefaultValue("/graphql") val path: String,
-    @DefaultValue val graphiql: DgsGraphiQLConfigurationProperties,
-    @DefaultValue val schemaJson: DgsSchemaJsonConfigurationProperties
+    @DefaultValue("/graphql") @field:ValidPath val path: String,
+    @DefaultValue @Valid val graphiql: DgsGraphiQLConfigurationProperties,
+    @DefaultValue @Valid val schemaJson: DgsSchemaJsonConfigurationProperties
 ) {
     /**
      * Configuration properties for the schema-json endpoint.
      */
     data class DgsGraphiQLConfigurationProperties(
         /** Path to the GraphiQL endpoint without trailing slash. */
-        @DefaultValue("/graphiql") val path: String
+        @DefaultValue("/graphiql") @field:ValidPath val path: String
     )
+
     /**
      * Configuration properties for the schema-json endpoint.
      */
     data class DgsSchemaJsonConfigurationProperties(
         /** Path to the schema-json endpoint without trailing slash. */
-        @DefaultValue("/schema.json") val path: String
+        @DefaultValue("/schema.json") @field:ValidPath val path: String
+    )
+
+    /**
+     * Convenience annotation to prevent having to repeat the @Pattern for every path field.
+     */
+    @kotlin.annotation.Target(AnnotationTarget.FIELD)
+    @kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
+    @Constraint(validatedBy = [])
+    @Pattern(regexp = "^/.*[^/]\$", message = "path must start with '/' and not end with '/'")
```

Comment:
_side note_, there is no [org.hibernate.validator.constraints](https://docs.jboss.org/hibernate/validator/7.0/api/) that already does this _path_ validation.